### ISSUE TITLE
features2d wrapper without flann module

### ISSUE THREAD

**Issue by dkurt** [opencv/opencv#10175]
##### System information (version)

- OpenCV => 3.3.1
- Operating System / Platform => Ubuntu 16.04
- Compiler => /usr/bin/c++ (ver 5.4.0)

##### Detailed description

[FlannBasedMatcher](https://github.com/opencv/opencv/blob/master/modules/features2d/include/opencv2/features2d.hpp#L1115) has two methods, constructor `FlannBasedMatcher` and `create()` that can't be wrapped without optional flann module but `CV_WRAP` ignores `#ifdef HAVE_OPENCV_FLANN`.

##### Steps to reproduce

```
cmake -DBUILD_LIST=features2d,python2 .. && make -j8
```

**Comment by wykvictor** [comment]
Any ideas how to solve it?  OpenCV == 3.3.1

**Comment by dkurt** [comment]
@wykvictor, For now only adding `flann` module to build:
```
cmake -DBUILD_LIST=features2d,flann,python2 .. && make -j8
```

**Comment by wykvictor** [comment]
@dkurt Thanks! Works for me.

**Comment by asmorkalov** [comment]
Re-targeted to 4.8.0 as OpenCV team does not produce OpenCV-Python releases from 3.4 branch any more.

**Comment by dkurt** [comment]
I will test with https://github.com/opencv/opencv/pull/16061

**Comment by asmorkalov** [comment]
@dkurt Have you had a chance to look at it?

**Comment by dkurt** [comment]
@asmorkalov, yes. Unfortunately, now only `#if 0` directive supported, not a conditional complication. I think this issue can be removed from 4.8.0 milestone.
